opm/parser/eclipse/EclipseState/Grid/GridDims.hpp

```diff
@@ -0,0 +1,174 @@
+/*
+  Copyright 2016  Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_PARSER_GRIDDIMS_HPP
+#define OPM_PARSER_GRIDDIMS_HPP
+
+#include <array>
+#include <stdexcept>
+#include <vector>
+
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/Deck/DeckKeyword.hpp>
+#include <opm/parser/eclipse/Deck/DeckRecord.hpp>
+#include <opm/parser/eclipse/Deck/Section.hpp>
+
+#include <opm/parser/eclipse/Parser/MessageContainer.hpp>
+#include <opm/parser/eclipse/Parser/ParserKeywords/D.hpp> // DIMENS
+#include <opm/parser/eclipse/Parser/ParserKeywords/S.hpp> // SPECGRID
+
+namespace Opm {
+    class GridDims
```

Comment:
When there is no templating involved I very much prefer a `GridDims.hpp` & `GridDims.cpp` pair.